File: src/main/java/com/ptaf/ai/audit/AiGenerationAuditRecord.java
package com.ptaf.ai.audit;

import java.util.List;

public record AiGenerationAuditRecord(
        String requestId,
        String timestamp,
        String command,
        String mode,
        String model,
        String promptVersion,
        String requirementHash,
        String outputPath,
        boolean parseSuccessful,
        boolean stepValidationPassed,
        boolean yamlValidationPassed,
        boolean fileWritten,
        int reusedStepsCount,
        int newStepsCount,
        int missingYamlKeysCount,
        List<String> warnings,
        List<String> blockingErrors
) {
}

File: src/main/ptaf/ai/audit/ai_generation_audit_record.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class AiGenerationAuditRecord:
    request_id: str
    timestamp: str
    command: str
    mode: str
    model: str
    prompt_version: str
    requirement_hash: str
    output_path: str
    parse_successful: bool
    step_validation_passed: bool
    yaml_validation_passed: bool
    file_written: bool
    reused_steps_count: int
    new_steps_count: int
    missing_yaml_keys_count: int
    warnings: list[str]
    blocking_errors: list[str]
